extensions/amp-image-slider/0.1/amp-image-slider.js

```diff
@@ -0,0 +1,676 @@
+          user().assertNumber(value,
+              'value to seek to must be a number');
+        } else if (args['percent'] !== undefined) {
+          value = args['percent'];
+          user().assertNumber(value,
+              'percent to seek to must be a number');
+          value *= 0.01; // percentage to 0-1 value
+        }
+        if (value !== undefined) {
+          this.updatePositions_(value);
+        }
+      }
+    }, ActionTrust.LOW);
+
+    return Promise.resolve();
+  }
+
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    const newDisabled = mutations['disabled'];
+    if (newDisabled) {
+      this.disable_(); // this.disabled_ is set in this call
+    } else {
+      this.enable_(); // this.disabled_ is set in this call
+    }
+  }
+
+  /**
+   * Enable user interaction
+   */
+  enable_() {
+    if (!this.disabled_) {
+      return;
+    }
+    // Show hint if needed
+    if (!this.disableHint_) {
+      this.animateShowHint_();
+    }
+    this.registerEvents_();
+    this.disabled_ = false;
+  }
+
+  /**
+   * Disable user interaction
+   */
+  disable_() {
+    if (this.disabled_) {
+      return;
+    }
+    // Hide hint if needed
+    if (!this.disableHint_) {
+      // need to clear timeout handle inside
+      // thus not directly calling animateHideHint_
+      this.resetHintInterval_(true); // no restart
+      this.isHintHidden_ = true;
+    }
+    this.unregisterEvents_();
+    this.disabled_ = true;
+  }
+
+  /**
+   * Build images
+   */
+  buildImages_() {
+    // Hierarchy:
+    // leftMask
+    //   |_ leftAmpImage
+    // rightMask
+    //   |_ rightAmpImage
+    this.container_.appendChild(this.rightMask_);
+    this.container_.appendChild(this.leftMask_);
+
+    this.rightMask_.appendChild(this.rightAmpImage_);
+    this.rightMask_.classList.add('i-amphtml-image-slider-right-mask');
+
+    if (this.rightLabel_) {
+      this.rightLabelWrapper_ = this.doc_.createElement('div');
+      this.rightLabelWrapper_.classList
+          .add('i-amphtml-image-slider-right-label-wrapper');
+      this.rightLabel_.classList.add('i-amphtml-image-slider-right-label');
+      this.rightLabelWrapper_.appendChild(this.rightLabel_);
+      this.rightMask_.appendChild(this.rightLabelWrapper_);
+    }
+    this.leftMask_.appendChild(this.leftAmpImage_);
+
+    this.leftMask_.classList.add('i-amphtml-image-slider-left-mask');
+    this.leftAmpImage_.classList.add('i-amphtml-image-slider-over-image');
+
+    if (this.leftLabel_) {
+      this.leftLabelWrapper_ = this.doc_.createElement('div');
+      this.leftLabelWrapper_.classList
+          .add('i-amphtml-image-slider-left-label-wrapper');
+      this.leftLabel_.classList.add('i-amphtml-image-slider-left-label');
+      this.leftLabelWrapper_.appendChild(this.leftLabel_);
+      this.leftMask_.appendChild(this.leftLabelWrapper_);
+    }
+  }
+
+  /**
+   * Build bar
+   */
+  buildBar_() {
+    this.container_.appendChild(this.bar_);
```

Comment:
Same comment as on the 0.0 version, we should try to to avoid reflow as much as possible. =)